### ISSUE TITLE
[Javascript] Implement new Concept Exercise: Basic Arrays

### ISSUE THREAD

**Issue by ovidiu141** [exercism/v3#944]
This issue describes how to implement the `basic_arrays` concept exercise for the Javascript track.

## Getting started

**Please please please read the docs before starting.** Posting PRs without reading these docs will be a lot more frustrating for you during the review cycle, and exhaust Exercism's maintainers' time. So, before diving into the implementation, please read up on the following documents:

- [The features of v3](https://github.com/exercism/v3/blob/master/docs/concept-exercises.md).
- [Rationale for v3](https://github.com/exercism/v3/blob/master/docs/rationale-for-v3.md).
- [What are concept exercise and how they are structured?](https://github.com/exercism/v3/blob/master/docs/features-of-v3.md)

Please also watch the following video:

- [The Anatomy of a Concept Exercise](https://www.youtube.com/watch?v=gkbBqd7hPrA).

## Goal

The goal of this exercise is to teach the student the basics of the Concept of Arrays in Javascript.

## Learning objectives

- Know of the existence of arrays.
- Know how to create an array.
- Know how to access values from an array.
- Know how to add and remove values from an array (`splice `and `pop`)
- Know how to get the length of an array

## Out of scope

- Looping over an array.
- Any array methods, except `pop`, `splice` and `push`. _e.g. `filter`, `reduce`, `sort`, `reverse` etc_
- Spread Syntax

## Concepts

TBD

## Prerequisites

`sameness`
`conditionals`
`looping`

## Representer

This exercise does not require any specific representation logic to be added to the [representer](https://github.com/exercism/javascript-representer).

## Analyzer

This exercise does not require any specific logic to be added to the [analyzer](https://github.com/exercism/javascript-analyzer).

## Implementing

Porting it from [Go's basic-slices](https://github.com/exercism/v3/tree/master/languages/go/exercises/concept/basic-slices).

## Help

If you have any questions while implementing the exercise, please post the questions as comments in this issue.

**Comment by Calamari** [comment]
Why is looping a prerequisite if looping over an array is out of scope?

**Comment by mindit-ovidiumiu** [comment]
> Why is looping a prerequisite if looping over an array is out of scope?

Well, one task of the exercise is building an array of cards (see #1033), so the student will probably use a `for/while` loop, but this is not really looping over an existing array (by using `forEach` or using `for` along with `length` and accessing an element by its index). At least this is how I see it.

**Comment by ovidiu141** [comment]
> Why is looping a prerequisite if looping over an array is out of scope?

Well, one task of the exercise is building an array of cards (see #1033), so the student will probably use a for/while loop, but this is not really looping over an existing array (by using forEach or using for along with length and accessing an element by its index). At least this is how I see it.

**Comment by SleeplessByte** [comment]
That's interesting. I think it will work fine, for now. We can change the prereqs later if need be :)
